tools/tmt/README.md

```diff
@@ -0,0 +1,135 @@
+maps, as defined above.
+
+Optionally, the location path may be followed by a single parameter
+specifying a .NET regular expression to match Managed and Java types
+in order to filter the map (see [Filtering](#type-name-filtering) below)
+
+## Reporting 
+
+By default, the utility generates files with full contents of type
+maps read from the shared library.  The output differs depending on
+whether the `Release` or `Debug` configuration is found in the shared
+library.
+
+For `Release`, two files for each native shared library are generated:
+
+  * Raw output (.raw)
+    Contains uninterpreted raw data from the map.  This is the most
+    technical, detailed format.  It performs no type name lookup,
+    contains information about managed assemblies, their modules,
+    identifiers etc.
+  * Processed output (.txt)
+    Contains a comma-separated value collection of type mappings, with
+    most of the technical details of the raw format removed, managed
+    type names looked up (if possible).  There are two modes for this
+    format - full and abbreviated.  They differ in that the latter
+    does not contain information about Managed type module MVID or its
+    token ID.
+
+For `Debug`, only one file is generated with the same information as
+the `Processed output` format above.
+
+In both cases, file names are constructed following the pattern below:
+
+    typemap-vX-CONFIGURATION-ARCHITECTURE-KIND.{raw,txt}
+
+Where
+
+  * `X` stands for the type map format version
+  * `CONFIGURATION` is either `Debug` or `Release`
+  * `ARCHITECTURE` is one of: `ARM`, `ARM64`, `X86`, `X86_64`
+  * `KIND` is either `managed` or `java`
+
+If regular expression type name filtering is not enabled, the utility
+will not print the map contents to the console.  This is because even
+small applications can contain thousands of entries and the whole map
+content is better written to a file instead of producing voluminous
+output on the screen.
+
+## Filtering
+
+### Type name filtering
+
+Passing a regular expression to the utility causes it to match type
+names (both Managed and Java for **each** loaded entry, regardless on
+whether the current map being processed is java-to-managed or
+managed-to-java) against it and report the matches to the console.
+
+Also, in this case the files are not generated by the default, which
+can be changed by using the `-g` option.  This is because filtering
+mode is mostly useful for "quick look" at the maps and the full output
+will likely be of no interest in this instance.
+
+### Architecture filtering
+
+By default the utility reads `libxamarin-app.so` for any architecture
+it finds in the location indicated on command line.  The `-a` option
+can be used to limit the set of architectures by specifying a
+comma-separated list of architecture names from the following set
+(architecture names are case-insensitive): `ARM`, `ARM64`, `X86` and
+`X86_64`
+
+# Command line options
+
```

Comment:
This should use a triple-backtick "block" so that it renders as intended.

Please provide an example invocation, e.g. on `samples/HelloWorld` output.

Reply:
Done